ci/process_notebooks.py

```diff
@@ -37,6 +42,167 @@
 )
 
 
+class NMAPreprocessor(ExecutePreprocessor):
+    """
+    Custom subclass of the ExecutePreprocessor for NMA tutorials.
```

Comment:
Can you add a link to the source of ExecutePreprocessor?

Reply:
Done